### PR TITLE
Flexlayout karma raj

### PR DESCRIPTION
**Pull request by rajgthub**
This is my answer for the homework. Please check and let me know if I need to do anything (@LucyMac ). I have  not considered responsive design at this moment.

### CODE REVIEW

**Comment by rajgthub** [comment]
Thank you very much for the positive comments. Yes, I can fix it.

**Comment by rajgthub** [comment]
I am unsure whether I made the pull request correctly! I think I have done my best to fix navigation section and the layout for the cases considering small devices. I have to find a way to make my navigation  collapse. Thanks for spending your valuable time on  checking my code and giving feedback to improve my knowledge further.

File: css/_abstract.scss [hunk 1]
```diff
@@ -7,4 +7,38 @@ $blk-col:#000000;
...
+        @media (max-width: 75em) {//1200px/16
+            @content
+        };
+    }
+    @if $break-point == big-desk {
+        @media (min-width: 112.5em) {//1800px/16
+            @content
+        };
+    }
+
```
Comment (LucyMac) [inline css/_abstract.scss]: This seems like overkill Raj, I would try something simpler. If you want to store your breakpoints inside variables, that's good practice. For example:
```
$max-width-mobile: 767px;

@media (max-width: $max-width-mobile) {
    margin-left: 0;
  }
``` 
this is more semantic than `@include respond(phone) {`
With `@include respond(phone) {` I do not know if we are talking about a size smaller than a phone or bigger than a phone. If you use the term `max-` or `min-` it's easier to understand what's going on 🙂

File: css/style.css [hunk 2]
```diff
@@ -432,18 +440,33 @@ th {
...
 
 .container {
-  width: 60%;
+  width: 65%;
   display: flex;
   justify-content: space-between;
 }
 
+@media (max-width: 56.25em) {
+  .container {
```
Comment (LucyMac) [inline css/style.css]: I find that using `em` for breakpoints makes it harder to understand. Consider using pixels which are more predictable and are widely used in media queries.

File: css/style.css [hunk 3]
```diff
@@ -432,18 +440,33 @@ th {
...
   display: flex;
   justify-content: center;
   align-items: center;
 }
 
+@media (max-width: 37.5em) {
+  .navigation {
+    opacity: 0;
+    visibility: hidden;
+  }
```
Comment (LucyMac) [inline css/style.css]: I suppose you did this in anticipation of collapsing your menu? 🙂

File: css/_abstract.scss (hunk 1 above)
Comment (LucyMac) [reply css/_abstract.scss]: I'm not sure the `@if` statements are necessary either. Keep it simple!

File: css/style.css [hunk 4]
```diff
@@ -542,12 +593,31 @@ button:hover {
...
   list-style: none;
   display: flex;
   justify-content: space-between;
 }
 
+@media (max-width: 37.5em) {
+  .cases__devices {
+    flex-direction: column;
+  }
+}
```
Comment (LucyMac) [inline css/style.css]: Good, this looks great! 

Watch out for the spacing around that section - there is no padding at the top and around the headline, and a lot of whitespace at the bottom. Try and even the space out.

File: css/style.css [hunk 5]
```diff
@@ -542,12 +593,31 @@ button:hover {
   align-self: center;
 }
 
+@media (max-width: 56.25em) {
+  .cases__content > h2 {
+    font-size: 2rem;
+    margin: 0;
+  }
```
Comment (LucyMac) [inline css/style.css]: Give this headline some breathing space 🌬

Comment (rajgthub) [reply css/style.css]: Yes, surely!

File: css/style.css (hunk 4 above)
Comment (rajgthub) [reply css/style.css]: Yes, I will add padding.

File: css/style.css (hunk 3 above)
Comment (rajgthub) [reply css/style.css]: Yes, that is right. I will fix it later.

File: css/style.css (hunk 2 above)
Comment (rajgthub) [reply css/style.css]: I read something about using px and rem in media query in the link: https://zellwk.com/blog/media-query-units/
If you read the following on the page: (2 and 3)
2. User Zooms In
The original idea behind em based sizes was due to older browsers not being able to update pixel values when a user zooms. In this regard, testing the difference between media query units when a user zooms will help to answer the question on whether we can use px based media queries now.
3. User Changed Their Browser’s Font Setting

Concluding The Experiments
As you can see from our tests above, the only unit that performed consistently across all four browsers is em. There aren’t any differences between em and rem with the exception of bugs found on Safari.

px media queries performed well in two of the three experiments (with the exception of Safari, again). Unfortunately, px media queries remained at 400px in the third experiment, which makes it a no-go if you intend to support users who change their browser’s font-size value.

Hence, my conclusion after these experiments is: Use em media queries.

File: css/_abstract.scss (hunk 1 above)
Comment (rajgthub) [reply css/_abstract.scss]: For this simple case, it is not necessary.

**Comment by rajgthub** [comment]
I have added more changes to the code. Please have a look at them when you get time. Thanks very much!

**Pull request closed without merging**